python/hsfs/storage_connector.py

```diff
@@ -927,3 +928,97 @@ def read_stream(
             options,
             include_metadata,
         )
+
+
+class GcsConnector(StorageConnector):
+    type = StorageConnector.GCS
+
+    def __init__(
+        self,
+        id,
+        name,
+        description,
+        featurestore_id,
+        # members specific to type of connector
+        key_path=None,
+        algorithm=None,
+        encryption_key=None,
+        encryption_key_hash=None,
+    ):
+        super().__init__(id, name, description, featurestore_id)
+
+        self._key_path = key_path
+        self._algorithm = algorithm
+        self._encryption_key = encryption_key
+        self._encryption_key_hash = encryption_key_hash
+
+    @property
+    def key_path(self):
+        """GCS key file HDFS path"""
+        return self._key_path
+
+    @property
+    def algorithm(self):
+        """Encryption Algorithm"""
+        return self._algorithm
+
+    @property
+    def encryption_key(self):
+        """Encryption Key"""
+        return self._encryption_key
+
+    @property
+    def encryption_key_hash(self):
+        """Encryption Key Hash"""
+        return self._encryption_key_hash
+
+    def spark_options(self):
+        """Return prepared options to be passed to Spark, based on the additional
+        arguments.
+        """
+        return super().spark_options
```

Comment:
not sure why you need this, it should inherit the method from the base class already.

Reply:
I was getting exceptions after removing this method so added it to return a blank dictionary